kernel/kernel_drivers.go

```diff
@@ -122,16 +122,38 @@ func createFirmwareSymlinks(orig, dest string) error {
 	return nil
 }
 
-func createModulesSubtree(origin, dest string) error {
-	kversion, err := KernelVersionFromModulesDir(origin)
-	if err != nil {
-		// Bit of a corner case, but maybe possible. Log anyway.
-		// TODO detect this issue in snap pack, should be enforced
-		// if the snap declares kernel-modules components.
-		logger.Noticef("no modules found in %q", origin)
+func setupModsFromComp(dest, kversion string, compInfos []*KernelModulesMntInfo) error {
+	// This folder needs to exist always to allow for directory swapping
+	// in the future, even if right now we don't have components.
+	compsRoot := filepath.Join(dest, "lib", "modules", kversion, "updates")
+	if err := os.MkdirAll(compsRoot, 0755); err != nil {
+		return err
+	}
+
+	if len(compInfos) == 0 {
 		return nil
 	}
 
+	// Symbolic links to components
+	for _, ci := range compInfos {
+		lname := filepath.Join(compsRoot, ci.CompSideInfo.Component.ComponentName)
+		to := filepath.Join(ci.MntDir, "modules", kversion)
+		if err := osSymlink(to, lname); err != nil {
+			return err
+		}
+	}
+
+	// Run depmod
+	stdout, stderr, err := osutil.RunSplitOutput("depmod", "-b", dest, kversion)
+	if err != nil {
+		return osutil.OutputErrCombine(stdout, stderr, err)
+	}
+	logger.Noticef("depmod output:\n%s\n", string(osutil.CombineStdOutErr(stdout, stderr)))
+
+	return nil
+}
+
+func createModulesSubtree(origin, dest, kversion string, compInfos []*KernelModulesMntInfo) error {
```

Comment:
I wonder if we could fine slightly more descriptive names than origin/dest given we kind of know how we are using things